GoBoard.java

```diff
@@ -0,0 +1,139 @@
+import java.awt.*;
+import java.awt.event.*;
+import java.util.Random;
+import javax.swing.*;
+
+public class GoBoard{
+    public static void main(String[] args) {
+	// create a new frame to hold the panel
+        JFrame application = new JFrame();
+        // set the layout of the frame  
+        application.setLayout(new BorderLayout());
+	// add the panel to the frame
+        application.add(new GoPanel(19), BorderLayout.CENTER);
+		
+        application.setSize(600, 625);
+	// set the frame to exit when it is closed
+        application.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	// make sure the size of the frame does not change
+        application.setResizable(false);
+	// make the frame visible
+        application.setVisible(true);
+    }
+}
+
+class GoPanel extends JPanel {
+
+    Square[][] board;
+    boolean whiteToMove;
+
+    //	GoPanel Constructor
+    GoPanel(int dimension) {
+        board = new Square[dimension][dimension];
+        whiteToMove = true;
+        initBoard(dimension);
+    }
+
+	// a Method that initialises the Board of the game 
+    private void initBoard(int dimension) {
+        super.setLayout(new GridLayout(dimension, dimension));
+        for(int row = 0; row < dimension; row++) {
+            for(int col = 0; col < dimension; col++) {
+                board[row][col] = new Square(row, col);
+                super.add(board[row][col]);
+            }
+        }
+        repaint();
+    }
+
+	// a method to place the stones in the intersections using a mouse click and paints/draw the lines on the board
+    private class Square extends JPanel {
```

Comment:
So far we have a GoBoard class, a GoPanel class and a Square class, all trying to accomplish one thing. Furthermore, GoPanel and Square both extend JPanel, I think two of these classes can be eliminated.